gcloud/pubsub/test_subscription.py

```diff
@@ -620,29 +603,68 @@ def test_check_iam_permissions_w_alternate_client(self):
 
         self.assertEqual(len(allowed), 0)
         self.assertEqual(len(conn1._requested), 0)
-        self.assertEqual(len(conn2._requested), 1)
-        req = conn2._requested[0]
-        self.assertEqual(req['method'], 'POST')
-        self.assertEqual(req['path'], PATH)
-        self.assertEqual(req['data'], REQUESTED)
+        self.assertEqual(len(conn2._requested), 0)
+        self.assertEqual(conn2._tested_iam_permissions,
+                         (self.SUB_PATH, ROLES))
 
 
-class _Connection(object):
+class _Connection(object):  # pylint: disable=too-many-instance-attributes
```

Comment:
1. The [over-ride][1] I linked to above is not the global config, it is the test config
1. We almost never run up against the limits for real reasons, just arbitrary code like this which could be written 10 different ways
1. Here we **are** mapping an API under our control, namely our implementation
1. You could deal with the too many attributes just by using `_responses` and scoping tests so that they don't cross paths on which methods use the `_responses`. (We did this before when `api_request` was the only consumer of `_responses`.)
1. Logilab is no longer the steward, `pylint` is community owned

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/ac50331bebc0fa92071d7e4fe9bdd66b139f1964/scripts/run_pylint.py#L61-L70